model_analyzer/triton/server/server_handler.py

```diff
@@ -0,0 +1,199 @@
+# Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from model_analyzer.triton.server.server_config import TritonServerConfig
+from model_analyzer.triton.server.server_factory import TritonServerFactory
+
+from model_analyzer.config.input.config_utils import binary_path_validator
+
+from model_analyzer.constants import CONFIG_PARSER_FAILURE, LOGGER_NAME
+
+from urllib.parse import urlparse
+
+import os
+import logging
+
+logger = logging.getLogger(LOGGER_NAME)
+
+
+class TritonServerHandler():
+    """
+    Static methods used to create and return Triton Servers
+    """
+
+    def __init__(self):
+        NotImplemented
+
+    @staticmethod
+    def get_server_handle(config, gpus, use_model_repository=False):
+        """
+        Creates and returns a TritonServer
+        with specified arguments
+
+        Parameters
+        ----------
+        config : namespace
+            Arguments parsed from the CLI
+        gpus : list of str
+            Available, supported, visible requested GPU UUIDs
+        use_model_repository : bool
+            Optional flag to use the model directory instead
+            of the output model directory
+        Returns
+        -------
+        TritonServer
+            Handle to the Triton Server
+        """
+
+        if config.triton_launch_mode == 'remote':
+            triton_config = TritonServerConfig()
+            triton_config.update_config(config.triton_server_flags)
+            triton_config['model-repository'] = 'remote-model-repository'
+            logger.info('Using remote Triton Server')
+            server = TritonServerFactory.create_server_local(
+                path=None, config=triton_config, gpus=[], log_path="")
```

Comment:
I think this line shows us that the GPUs is optional. As such, you could make the default "[ ]" and not have to plumb gpus everywhere.  That should allow you drop the number of files changed significantly.